### DISCUSSION TITLE
The feature about is_cached

### DISCUSSION THREAD

**Discussion by happyych** [fumitoh/modelx#167] [Q&A]
Hello,

I have a question about using is_cached  option. 
When I read about the option in tutorial, I think it is super cool, because I wanted to reference two or more points in time for a variable simultaneously.

When I tried the option, I faced an error messege as below.
I guess setting is_cached option as False doesn't work in vectorized model.

Does the is_cached option also works in vectorized model? 

If I have to set the option in other way, pleas let me know.

Thank you always!

**Comment by fumitoh** [comment]
Have you updated modelx to v0.27.0?

**Reply by happyych** [reply]
Yes I tried the code above based on modelx v0.27.0

![version](https://github.com/user-attachments/assets/f1504b6c-47fc-41a1-90a1-08e3cd8b1393)

**Comment by fumitoh** [comment]
Thank you. Judging from the traceback message, the error seems to be raised when trying to print the error message for another error.
So `RENEWAL([1, 2])` is failing already. I think passing the tuple `(1, 2)` instead of  the list shows the traceback of the original error. 
I'll fix the error during outputting the error message.

**Reply by happyych** [reply]
Thank you for the fast reply. I tried tuple as your recomendation.

**Reply by fumitoh** [reply]
Yes, this is a formula error, so the ValueError message is indicating there's an error around a shape mismatch of numpy arrays.
I think the error message should show what line in `RENEWAL` the error occured although not captured in the screenshot.

**Reply by fumitoh** [reply]
Yes, so as the errormessage says, you can't compare a tuple with 2 elements and an array with 5 elements. 
I tested outside modelx:
```python
import numpy as np
t = (1, 2)
DURATIONIF_TO_CVRG = np.array([0.,0.,0.,0.,0.])
POL_TERM_U = np.array([1.,1.,1.,1.,1.])
(t <= 1 + DURATIONIF_TO_CVRG | ((t - DURATIONIF_TO_CVRG -1) % (POL_TERM_U * 12) != 0))
```
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[52], line 5
      3 DURATIONIF_TO_CVRG = np.array([0.,0.,0.,0.,0.])
      4 POL_TERM_U = np.array([1.,1.,1.,1.,1.])
----> 5 (t <= 1 + DURATIONIF_TO_CVRG | ((t - DURATIONIF_TO_CVRG -1) % (POL_TERM_U * 12) != 0))

ValueError: operands could not be broadcast together with shapes (2,) (5,) 
```

**Reply by happyych** [reply]
Sorry for the late reply. It takes some time to figure out the error of my code. Now I understand the situation and feature of is_cached    feature. Thank you for your kind explanation!
